Fix crda

francois-berder-imgtec opened the pull request:
These patches fix cfg80211 calling CRDA at boot and failing to do so. In openwrt, crda package has been removed. Instead, cfg80211 must hold an internal database. We were building as built-in cfg80211 (configured without internal crda regdb) and mac80211 from the linux kernel and kmod-cfg80211 (with internal regdb) as a module. The module was never used and could not even be loaded because it would duplicate some symbols with the built-in cfg80211 from the linux kernel.
The solution consists in using cfg80211 and mac80211 from the linux kernel exclusively and configures them properly such that they use an internal database.

francois-berder committed:
Add CRDA database patch

The default CRDA database is empty, replacing it with
a copy from wireless-regdb repository.

Signed-off-by: Francois Berder <Francois.Berder@imgtec.com>

Avinash95 commented on target/linux/pistachio/config-4.1:
```diff
@@ -46,6 +46,13 @@ CONFIG_CC_STACKPROTECTOR=y
 CONFIG_CC_STACKPROTECTOR_STRONG=y
 CONFIG_CEVT_R4K=y
+CONFIG_CFG80211=y
+CONFIG_CFG80211_DEBUGFS=y
```
Do we need this ?

nikhil-zinjurde-imgtec commented:
test with ca8210 profile

abhijit-mahajani-imgtec commented:
@datachi7d , i remember our conversation in #8, as well as #20, will this be fine to enable these modules in kernel again since its fixing the CRDA issue?

francois-berder-imgtec commented:
To add more information:
kmod-cfg80211 belongs to compat-wireless whereas cfg80211 belongs to the linux kernel. I could build kmod-cfg80211 and kmod-mac80211 as built-in and cfg80211 and mac80211 as modules. This was the only other way to build uccp420wlan driver. However, loading the wifi driver failed to find ieee80211_rx symbol which is defined by mac80211. I suspect that since it was getting linked with mac80211, it could not work with kmod-mac80211.

nikhil-zinjurde-imgtec commented:
test with ca8210 profile

abhijit-mahajani-imgtec commented:
Not getting wlan0 in the ifconfig when tried this change after cleaning up target/linux and rebuilding again..

francois-berder-imgtec commented:
@abhijit-mahajani-imgtec Indeed, I rebased these patches on top of master-pistachio, did a clean build and wlan0 is not appearing in ifconfig.

francois-berder committed:
modules.mk: configure cfg80211 to use internal regdb

On Openwrt, crda package has been removed. So cfg80211 must use
an internal database.

Signed-off-by: Francois Berder <Francois.Berder@imgtec.com>

francois-berder-imgtec commented:
I found out that if I did not build kmod-cfg80211 and kmod-mac80211, the wifi driver would not load properly.

nikhil-zinjurde-imgtec commented:
test with ca8210 profile

nikhil-zinjurde-imgtec commented:
Verified, CRDA logs are no longer displayed on OpenWrt prompt
